ibm/resource_ibm_cis_healthcheck.go

```diff
@@ -2,331 +2,364 @@ package ibm
 	if err != nil {
-		if checkCisMonitorDeleted(d, meta, err, monitor) {
-			d.SetId("")
-			return nil
-		}
-		log.Printf("[WARN] Error getting zone during MonitorRead %v\n", err)
+		log.Printf("Error reading global load balancer health check detail: %s", resp)
 		return err
 	}
-	monitorObj := *monitor
-	d.Set("cis_id", cisId)
-	d.Set("description", monitorObj.Description)
-	d.Set("path", monitorObj.Path)
-	d.Set("expected_body", monitorObj.ExpBody)
-	d.Set("expected_codes", monitorObj.ExpCodes)
-	d.Set("type", monitorObj.MonType)
-	d.Set("method", monitorObj.Method)
-	d.Set("timeout", monitorObj.Timeout)
-	d.Set("retries", monitorObj.Retries)
-	d.Set("interval", monitorObj.Interval)
-	d.Set("follow_redirects", monitorObj.FollowRedirects)
-	d.Set("allow_insecure", monitorObj.AllowInsecure)
-	d.Set("port", monitorObj.Port)
+	d.Set(cisGLBHealthCheckID, result.Result.ID)
+	d.Set(cisID, crn)
+	d.Set(cisGLBHealthCheckDesc, result.Result.Description)
+	d.Set(cisGLBHealthCheckPath, result.Result.Path)
+	d.Set(cisGLBHealthCheckExpectedBody, result.Result.ExpectedBody)
+	d.Set(cisGLBHealthCheckExpectedCodes, result.Result.ExpectedCodes)
+	d.Set(cisGLBHealthCheckType, result.Result.Type)
+	d.Set(cisGLBHealthCheckMethod, result.Result.Method)
+	d.Set(cisGLBHealthCheckTimeout, result.Result.Timeout)
+	d.Set(cisGLBHealthCheckRetries, result.Result.Retries)
+	d.Set(cisGLBHealthCheckInterval, result.Result.Interval)
+	d.Set(cisGLBHealthCheckFollowRedirects, result.Result.FollowRedirects)
+	d.Set(cisGLBHealthCheckAllowInsecure, result.Result.AllowInsecure)
+	d.Set(cisGLBHealthCheckPort, result.Result.Port)
+	d.Set(cisGLBHealthCheckCreatedOn, result.Result.CreatedOn)
+	d.Set(cisGLBHealthCheckModifiedOn, result.Result.ModifiedOn)
+
 	return nil
 }
 
-func resourceCIShealthCheckUpdate(d *schema.ResourceData, meta interface{}) error {
-	cisClient, err := meta.(ClientSession).CisAPI()
+func resourceCISHealthCheckUpdate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).CisGLBHealthCheckClientSession()
 	if err != nil {
 		return err
 	}
-	monitorId, cisId, err := convertTftoCisTwoVar(d.Id())
+
+	monitorID, crn, err := convertTftoCisTwoVar(d.Id())
 	if err != nil {
 		return err
 	}
-	monitorUpdate := v1.MonitorBody{}
-	if d.HasChange("type") || d.HasChange("description") || d.HasChange("path") || d.HasChange("expected_body") || d.HasChange("expected_codes") || d.HasChange("method") || d.HasChange("timeout") || d.HasChange("retries") || d.HasChange("interval") || d.HasChange("follow_redirects") || d.HasChange("allow_insecure") || d.HasChange("port") {
-		if monType, ok := d.GetOk("type"); ok {
-			monitorUpdate.MonType = monType.(string)
+	sess.Crn = core.StringPtr(crn)
+
+	opt := sess.NewEditLoadBalancerMonitorOptions(monitorID)
+	if d.HasChange(cisGLBHealthCheckType) ||
+		d.HasChange(cisGLBHealthCheckDesc) ||
+		d.HasChange(cisGLBHealthCheckPort) ||
+		d.HasChange(cisGLBHealthCheckExpectedCodes) ||
```

Comment:
should be expectedbody
`cisGLBHealthCheckExpectedCodes` repeated twice

Reply:
modified